### PR TITLE
Column identifiers (fields) must be enclosed in double quotes

### PR DESCRIPTION
**Pull request by TheGoldenEye**
additional: remove unused private.DOUBLE_DOUBLE_QUOTES definition

### CODE REVIEW

**Comment by TheGoldenEye** [comment]
this PR fixes #392

File: modules/sql.js [hunk 1]
```diff
@@ -100,6 +105,7 @@ __private.query = function (action, config, cb) {
 
 		try {
 			sql = jsonSql.build(extend({}, config, defaultConfig));
+			library.logger.trace("sql.query: ", sql);
```
Comment (4miners) [inline modules/sql.js]: Can you change double quotes to single? Tests failed because of that.

Comment (TheGoldenEye) [inline modules/sql.js]: What do you mean? Postgres does not accept single quotes in the column names enumeration

Comment (4miners) [inline modules/sql.js]: Ok, thanks. :)

**Comment by TheGoldenEye** [comment]
Changes done

**Comment by 4miners** [comment]
Ok, however after looking closer I cannot allow that PR to mege. Looks like those column names are user input, so we cannot just pass it to DB and have hope that there will be no pigs inside. We should handle that with proper care. I propose one of following:
- regex test and allow only specified chars `a-zA-Z`, `_` and `0-9` (but not at beginning)
- get column names from DB and compare with user input
- use format (https://www.postgresql.org/docs/current/static/plpgsql-statements.html#PLPGSQL-QUOTE-LITERAL-EXAMPLE), there is `pgp.as.format` function in core that we probably can use

**Comment by TheGoldenEye** [comment]
I cannot agree with you. The column names are not direct user input. 
Ok, the call comes from the sandboxAPI, so a dirty app can insert pigs in the query.
To prevent this, the column names and values are quoted. 
The column identifiers use __private.escape2(), the values use __private.escape()
This should work like quote_ident() and quote_literal(), as proposed by you.
Btw, with the sandboxAPI the apps can only modify the dapp tables like dapp_<dappid>_blocks.
There is a potential security issue, because no one prevents the app from modifying tables from other apps.
But this is another issue...

**Comment by 4miners** [comment]
Thanks for clarification. I tested it and your solution is valid for fields like `s!o%m?e'th"i;ng`.

**Pull request merged**